File: src/main/java/qmech/lib/objects/ToolTypeBase.java
package qmech.lib.objects;

import net.minecraft.item.Item.ToolMaterial;
import net.minecraft.item.ItemStack;
import net.minecraftforge.common.util.EnumHelper;

public class ToolTypeBase {
    public ToolTypeBase (String name, int harvestLevel, int durability, float efficiency, int damage, int enchantability, ItemStack material) {
        this.harvestLevel = harvestLevel;
        this.toolName = name;
        this.material = material;
        toolType = EnumHelper.addToolMaterial(name, harvestLevel, durability, efficiency, damage, enchantability);
    }

    public ToolTypeBase (String name, ToolInfo info, ItemStack material) {
        this.harvestLevel = info.harvestLevel;
        this.toolName = name;
        this.material = material;
        toolType = EnumHelper.addToolMaterial(name, info.harvestLevel, info.durability, info.efficiency, info.damage, info.enchantability);
    }


    public static class ToolInfo {
        public int harvestLevel = 1;
        public int durability = 400;
        public float efficiency = 2.0F;
        public int damage = 4;
        public int enchantability = 2;

        public ToolInfo() {}

        public ToolInfo(int harvestLevel, int durability, float efficiency, int damage, int enchantability) {
            this.harvestLevel = harvestLevel;
            this.durability = durability;
            this.efficiency = efficiency;
            this.damage = damage;
            this.enchantability = enchantability;
        }
    }
    
    ToolMaterial toolType;
    public ToolMaterial getToolType () {
        return toolType;
    }
    
    int harvestLevel;
    public int getHarvestLevel () {
        return harvestLevel;
    }
    
    String toolName;
    public String getToolName () {
        return toolName;
    }
    
    ItemStack material;
    public ItemStack getMaterial () {
        return material;
    }

    public static String[] shovelCraftingShape = new String[]{"x", "s", "s"};
    public static String[] pickaxeCraftingShape = new String[]{"xxx", " s ", " s "};
    public static String[] axeCraftingShape = new String[]{"xx","xs", " s"};
    public static String[] hoeCraftingShape = new String[]{"xx", " s", " s"};
    public static String[] swordCraftingShape = new String[]{"x", "x", "s "};
}

File: src/main/scala/geek/lawsof/physics/lib/equip/types/ToolTypeBase.scala
package geek.lawsof.physics.lib.equip.types

import geek.lawsof.physics.lib.CreativeTabBase
import geek.lawsof.physics.lib.equip._
import geek.lawsof.physics.lib.util.info.ToolInfo
import net.minecraft.item.Item.ToolMaterial
import net.minecraft.item.ItemStack
import net.minecraftforge.common.util.EnumHelper

/**
 * Created by anshuman on 28-05-2014.
 */
object ToolTypeBase {
  var shovelCraftingShape = Array("x", "s", "s")
  var pickaxeCraftingShape = Array("xxx", " s ", " s ")
  var axeCraftingShape = Array("xx", "xs", " s")
  var hoeCraftingShape = Array("xx", " s", " s")
  var swordCraftingShape = Array("x", "x", "s")
}

class ToolTypeBase(var toolName: String, var harvestLevel: Int, var durability: Int, var efficiency: Float, var damage: Int, var enchantability: Int, var craftMaterial: ItemStack) {
  toolType = EnumHelper.addToolMaterial(toolName, harvestLevel, durability, efficiency, damage, enchantability)

  def this(name: String, info: ToolInfo, mat: ItemStack) = this(name, info.harvestLevel, info.durability, info.efficiency, info.damage, info.enchantability, mat)

  var toolType = ToolMaterial.IRON

  val levels = Array("Stone", "Iron", "Diamond", "Obsidian")

  def toolLevel = if (harvestLevel < 4) levels(harvestLevel) else harvestLevel.toString

  var pickaxe: ToolPickaxe = null
  var shovel: ToolShovel = null
  var axe: ToolAxe = null
  var sword: ToolSword = null
  var hoe: ToolHoe = null

  def createToolSet(ctab: CreativeTabBase) = {
    pickaxe = new ToolPickaxe(this, ctab)
    shovel = new ToolShovel(this, ctab)
    axe = new ToolAxe(this, ctab)
    sword = new ToolSword(this, ctab)
    hoe = new ToolHoe(this, ctab)
  }
}
